### PR TITLE
vm/audit_log

### PR DESCRIPTION
**Pull request by dhvanilshah-tenable**
- Add support for audit_log endpoint in v3
- Added search method for v3 and doc string changes

### CODE REVIEW

**Review by harshshah-tenable — APPROVED** [review]

**Review by parvdesai-tenable — APPROVED** [review]

File: tenable/io/v3/vm/audit_log/api.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
+'''
+Audit Log
+=========
+
+The following methods allow for interaction into the Tenable.io
```
Comment (Mangusingh-te) [inline tenable/io/v3/vm/audit_log/api.py:5]:
```suggestion
The following methods allow interaction withTenable.io
```

File: tenable/io/v3/vm/audit_log/api.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                Examples:
+                    >>> [('field_name_1', 'asc'),
+                    ...      ('field_name_2', 'desc')]
+            limit (int, optional):
+                Number of objects to be returned in each request.
+                Default and max_limit is 200.
+            next (str, optional):
+                The pagination token to use when requesting the next page of
+                results. This token is presented in the previous response.
+            return_resp (bool, optional):
```
Comment (Mangusingh-te) [inline tenable/io/v3/vm/audit_log/api.py:93]: Update docstring.

File: tenable/io/v3/vm/audit_log/api.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                specific page of data.
+            return_csv (bool, optional):
+                If set to true, it will return the CSV response or
+                iterable (based on return_resp flag). Iterator returns all
+                rows in text/csv format for each call with row headers.
+        Returns:
+            Iterable:
+                The iterable that handles the pagination for the job.
+            requests.Response:
+                If ``return_json`` was set to ``True``, then a response
```
Comment (Mangusingh-te) [inline tenable/io/v3/vm/audit_log/api.py]: Update docstring.

File: tests/io/v3/vm/audit_log/test_audit_log_api.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+    iterator = api.v3.vm.audit_log.search(
+        fields=fields, filter=filter, sort=sort, limit=200
+    )
+    assert isinstance(iterator, SearchIterator)
+
+    event_list = []
+    for event in iterator:
+        event_list.append(event)
+    assert len(event_list) == api_res['pagination']['total']
```
Comment (Mangusingh-te) [inline tests/io/v3/vm/audit_log/test_audit_log_api.py]: Rewrite it in pythonic way.

**Review by Mangusingh-te — CHANGES_REQUESTED** [review]

File: tenable/io/v3/vm/audit_log/api.py (hunk 2 above)
Comment (alpesh-te) [reply tenable/io/v3/vm/audit_log/api.py:93]: Addressed

File: tests/io/v3/vm/audit_log/test_audit_log_api.py (hunk 4 above)
Comment (alpesh-te) [reply tests/io/v3/vm/audit_log/test_audit_log_api.py]: Addressed

File: tenable/io/v3/vm/audit_log/api.py (hunk 3 above)
Comment (alpesh-te) [reply tenable/io/v3/vm/audit_log/api.py]: Addressed

**Review by alpesh-te — APPROVED** [review]
LGTM!

**Pull request merged**